delegates.rb

```diff
@@ -83,93 +84,79 @@ def redirect(options = {})
   def authorize(options = {})
     logger = Java::edu.illinois.library.cantaloupe.script.Logger
    
-    u_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
+    g_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
     #'65.88.88.115'
     remote_ip = context['request_headers']['X-Forwarded-For']
     logger.debug("CONTEXT HASH: #{context}")
+    logger.debug("REQUEST HEADERS: #{context['request_headers']}")
+    logger.debug("CLIENT_IP: #{context['client_ip']}")
     logger.debug("IP ADDRESS: #{remote_ip}")
     logger.debug("REQUEST URI: #{context['request_uri']}")
-    # set type to variable since it will be referenced more frequently in future work
-    type = context['request_uri'].split('=')[1]
-    logger.debug("type: #{type}")
-    if context['request_uri'] =~ /ufile=true/
-      logger.debug("UFILE ACCESS")
-      if u_file_access.include?(remote_ip) || remote_ip =~ /^63.147.60./
+    split_image_url = context['request_uri'].split('/') # ie. ["http:", "", "localhost:8182", "iiif", "2", "1992268", "full", "!150,150", "0", "default.jpg"]
+    type = split_image_url[7]=="full" ? split_image_url[7] : derivative_type(context['resulting_size'])

```

Comment:
I don't think we need a split here. I think pass the resulting_size along to the look up method, and if it's greater than  2560, we can consider it full res (for rights purposes).